test/connectivity/modules/NetworkController/Program.cs

```diff
@@ -97,32 +94,34 @@ static async Task RemoveAllControllingRules(IList<INetworkController> controller
 
             foreach (var controller in controllerList)
             {
-                NetworkStatus status = await controller.GetStatusAsync(cancellationToken);
-                if (status != NetworkStatus.Default)
+                NetworkStatus networkStatus = await controller.GetNetworkStatusAsync(cancellationToken);
+                if (networkStatus != NetworkStatus.Disabled)
                 {
-                    Log.LogInformation($"Network is {status} with {controller.Description}, setting default");
-                    bool online = await controller.SetStatusAsync(NetworkStatus.Default, cancellationToken);
-                    online = await CheckSetStatusAsyncResult(online, NetworkStatus.Default, controller, cancellationToken);
+                    Log.LogInformation($"Network restriction is enabled for {controller.NetworkControllerType}. Setting default");
+                    bool online = await controller.SetNetworkStatusAsync(NetworkStatus.Disabled, cancellationToken);
+                    online = await CheckSetNetworkStatusAsyncResult(online, NetworkStatus.Disabled, controller, cancellationToken);
                     if (!online)
                     {
                         Log.LogError($"Failed to ensure it starts with default values.");
                         throw new TestInitializationException();
                     }
+                    else
```

Comment:
>else [](start = 20, length = 4)

this should be after the foreach as it was before because it has to check all rules before it reports the status